Added last_seen.js plugin

Lixquid opened the pull request:
Adds some "last seen" stats.

Bot will automatically welcome new people to the chat, and also report how long it's been since we've seen someone if they've visited before. If they had a different name, it will also report their old name.

Couple commands are included:

<del>`.lastseenid`: Looks up when a player was last connected via steamid</del> Use `.lastseen`.
`.lastseen`: Looks up when a player was last connected via name. Accepts SQLite wildcards (`%` and `_`). Limited to 10 results.

wiox commented:
> Should I maybe merge this with lastseen command and just do a regex check for a string of digits above a certain length?

It'd be cool to just change the SQL query in the lastseen command to use an OR operator and work with either input with the same code.

Lixquid commented:
Alright, removed the `.lastseenid` command.

I also fixed the "can't find players in chat" bug, it was me forgetting JavaScript can't handle big numbers. I had to change the db schema, so if you've tested this clientside or something, you'll need to drop it.

Otherwise this should be finished.

wiox commented:
What's up with wrapping it all in a function?

Lixquid commented:
Just stops my shorthand function naming from polluting the global namespace.

wiox commented:
It wouldn't do that anyway. Scoping in JavaScript is marginally different to Lua, in order for a variable to be set as global in Node.JS you've got to explicitly set it as a value in the "global" object.

Lixquid commented:
Ah, you're right. This'd be my browser js knowledge bleeding through.

Give me a sec, I'll strip them out.

wiox commented:
I think browsers execute code with the same "namespace". Node doesn't.

Lixquid commented:
Yeah, you're right.
